### PR TITLE
updated src for extended code generation

### PR DESCRIPTION
**Pull request by harikishants**

### CODE REVIEW

File: src/main/scala/daisy/backend/CodeGenerationPhase.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ object CodeGenerationPhase extends DaisyPhase {
   override val definedOptions: Set[CmdLineOption[Any]] = Set(
     StringChoiceOption(
       "lang",
-      Set("C", "Scala", "FPCore"),
+      Set("C", "OptimizedC", "Scala", "FPCore"),
```
Comment (dlohar) [inline src/main/scala/daisy/backend/CodeGenerationPhase.scala]: Please remove optimizedC from this file. For NN we have to use NNCodegen, not this one.

File: src/main/scala/daisy/opt/NNQuantizationPhase.scala [hunk 2]
```diff
@@ -239,7 +239,7 @@ object NNQuantizationPhase extends DaisyPhase with tools.RoundoffEvaluators with
 
     val f: Future[Unit] = Future {
       // Run soplex with the model until timeout
-      val res = Runtime.getRuntime().exec("/var/tmp/qnn/scipoptsuite-7.0.3/scip_build/bin/scip -f model.lp -s scip.set")
+      val res = Runtime.getRuntime().exec("/var/tmp/safe-qnn/scipoptsuite-7.0.3/scip_build/bin/scip -f model.lp -s scip.set")
```
Comment (dlohar) [inline src/main/scala/daisy/opt/NNQuantizationPhase.scala]: This also will not be changed. Revert it back to the original. I changed it to match the path for the machine we had.

File: src/main/scala/daisy/utils/CPrinter.scala [hunk 3]
```diff
@@ -430,6 +430,7 @@ class CPrinter(buffer: Appendable, ctx: Context) extends CodePrinter(buffer) {
         }
 
         sb.append(") {")
+        sb.append("\n#pragma HLS PIPELINE\n#pragma HLS ARRAY_PARTITION variable=_result complete")
         nl(lvl+1)
```
Comment (dlohar) [inline src/main/scala/daisy/utils/CPrinter.scala:512]: Where are the other changes in this file? I assume you had to edit for the nested code part as well.

**Review by dlohar — CHANGES_REQUESTED** [review]
I have added comments. Please address them so that we can merge this request. Also, you may want to change the name of the GitHub account that you are using. It shows Devleena Ghosh, not your name.

File: src/main/scala/daisy/utils/CPrinter.scala (hunk 3 above)
Comment (harikishants) [reply src/main/scala/daisy/utils/CPrinter.scala:512]: The nested code part is in separate file CPrinterOptimized.scala

**Comment by harikishants** [comment]
Are you able to see the CPrinterOptmized.scala?

**Comment by dlohar** [comment]
Ok. It looks good. I'm merging the src. For the next merge requests, please add a description so that people know what this merge request was about.

**Pull request merged**